Automattic/wp-calypso
# Gutenframe: Move Gutenboarding launch button to FSE

#### Changes proposed in this Pull Request

Move the _Launch_ button functionality from Gutenboarding to the FSE plugin.

This PR touches:
- Calypso
- wpcom-block-editor
- `full-site-editing` plugin

The changes need to be deployed in distinct steps. They've been designed to ensure that things do not conflict.

`wpcom-block-editor` currently handles adding the "Launch" button (#39048)

1. A change is made to `wpcom-block-editor` so that it won't handle the Launch button when FSE handling is triggered.
1. A change is made to FSE so that it will handle the Launch button when triggered.
1. **No changes in production at this point**
1. The Calypso iframe is changed to add the FSE trigger, in a single commit this will begin triggering FSE and not `wpcom-block-editor` handling.

#### Testing instructions

Sandbox a Simple site (`example.wordpress.com`) and `widgets.wp.com`.

1. Initially, don't use the Calypso code from this PR.
1. Apply D44230-code - you should see debug output like:
   ```
   WPCBE: Running update…
   WPCBE: aborting.
   WPCBE: Running update…
   WPCBE: handling…
   ```
   The launch button should be applied when you reach the block editor in Calypso from the [`/new`](https://wordpress.com/new) flow.
1. With the previous patch still applied, apply the FSE patch D44224-code. You should see 
   ```
   WPCBE: Running update…
   WPCBE: aborting.
   FSE: Running update…
   FSE: aborting.
   WPCBE: Running update…
   WPCBE: handling…
   FSE: Running update…
   FSE: aborting.
   ```
   The launch button should be applied when you reach the block editor in Calypso from the [`/new`](https://wordpress.com/new) flow.
1. The latest commit on this PR forces Gutenboarding in the ditor, so you can visit [`/block-editor/post`](https://calypso.live/block-editor/post?branch=fse/handle-launch-button) for your sandboxed site. You should now see FSE handling the launch button:
   ```
   WPCBE: Running update…
   WPCBE: aborting.
   FSE: Running update…
   FSE: aborting.
   WPCBE: Running update…
   WPCBE: aborting.
   FSE: Running update…
   FSE: handling…
   ```
1. Confirm that the whole from from this PR is working properly. We'll need to start from an earlier commit without the forced Gutenboarding: [`/new`](https://hash-28ebbf25c557049e4f4588351aa641c2132c4280.calypso.live/new). You should see the same behavior when you reach the block editor.

## Next steps

- [x] Extract a PR from d0374dd - https://github.com/Automattic/wp-calypso/pull/42849
     - [x] Commit/Deploy
- [x] Extract a PR from 4783580 - https://github.com/Automattic/wp-calypso/pull/42850
     - [x] Merge https://github.com/Automattic/wp-calypso/pull/42850
     - [x] FSE Version bump: #42895
     - [x] Commit/Deploy
     - [x] Release FSE plugin
     - [x] Wait until atomic is running new FSE version

_☝️  These two primary steps can, but at this point we should block until everything is complete. Then we can proceed to trigger the switchover_

- [x] Commit the switch in the Calypso Gutenberg iframe 28ebbf2 (https://github.com/Automattic/wp-calypso/pull/42931 / D44333-code)

Closes https://github.com/Automattic/wp-calypso/issues/42647
Enables work on https://github.com/Automattic/wp-calypso/issues/42655
